app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt

```diff
@@ -497,6 +512,20 @@ class SettingsFragment : Fragment(R.layout.activity_settings_screen) {
             }.start()
         }
 
+        b.settingsActivityNotificationRl.setOnClickListener {
+            b.settingsActivityNotificationRl.isEnabled = false
+            showDialogForNotificationAction()
+            object : CountDownTimer(500, 500) {
+                override fun onTick(millisUntilFinished: Long) {}
+
+                override fun onFinish() {
+                    if (isAdded) {
+                        b.settingsActivityNotificationRl.isEnabled = true
+                    }
+                }
+            }.start()
```

Comment:
this looks inefficient? what is it doing? Isn't there a better way to check `isAdded`?

Reply:
The countDownTimer is to disable the UI element for 500ms. And will be enabled after that. 
isAdded is the component of Fragment. 
Either we can use Handler or CountDownTimer so chose to use CountDownTimer instead of Handler.

Reply:
Mention that in the code comment?

```
// the count-down-timer ticks once after 500ms, at the completion of which
// the <ui-element-here> [is re-enabled again or add whatever is appropriate.
```

Reply:
Added comments for the UI elements.